src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/CachingCodeFixProviderForProjects.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Refactoring.V2
+{
+    [Shared]
+    [Export(typeof(CachingCodeFixProviderForProjects))]
+    public class CachingCodeFixProviderForProjects
+    {
+        private readonly ConcurrentDictionary<ProjectId, IEnumerable<CodeFixProvider>> _cache = new ConcurrentDictionary<ProjectId, IEnumerable<CodeFixProvider>>();
+        private readonly ILogger<CachingCodeFixProviderForProjects> _logger;
+
+        [ImportingConstructor]
+        public CachingCodeFixProviderForProjects(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<CachingCodeFixProviderForProjects>();
+        }
+
+        public IEnumerable<CodeFixProvider> GetAllCodeFixesForProject(ProjectId projectId)
+        {
+            if (_cache.ContainsKey(projectId))
+                return _cache[projectId];
+            return Enumerable.Empty<CodeFixProvider>();
```

Comment:
Why return empty instead of attempting to load the analyzers?

Reply:
Cache is filled when project is loaded (ProjectManager calls LoadFrom method). When CodeFixes are requested we don't have reference for ProjectInfo object so we don't have enough information to load code fixes at that point. ProjectInfo seems only be used until projects are loaded to workspace -> become project objects. However project doesn't have required information with it.